tests/tasks/manage_test_interface.yml

```diff
@@ -26,8 +26,27 @@
     - ip link set {{ interface }} up
   when: "type == 'veth' and state == 'present' and
          interface not in current_interfaces"
+
+- name: Install NetworkManager in CentOS7 when running with nm provider
+  package:
+    name: NetworkManager
+    state: present
+  when:
+    - ansible_distribution_major_version == '7' and
+      ansible_distribution == 'CentOS'
+    - network_provider == 'nm'
+
+- name: Start NetworkManager in CentOS7 when running with nm provider
+  command: systemctl start NetworkManager
+  when:
+    - ansible_distribution_major_version == '7' and
+      ansible_distribution == 'CentOS'
+    - network_provider == 'nm'
+
 - name: Set up veth as managed by NetworkManager
-  command: nmcli d set {{ interface }} managed true
+  shell: |
+    nmcli c reload
+    nmcli d set {{ interface }} managed true
```

Comment:
IMHO, the previous code here and the new code are both wrong. The role should/needs to be fixed to handle this instead of having the tests prepare the system to work. This will allow admins to use the network role to manager the IP configuration etc on veth interfaces.